src/components/Drawer/Drawer.js

```diff
@@ -1,55 +1,112 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
+import { Transition } from 'react-transition-group';
 import bem from 'bem';
 import { IoIosArrowUp } from 'react-icons/io';
 import styles from './Drawer.scss';
 
+const TRANSITION_DURATION = {
+    enter: 10,
+    exit: 300,
+};
+
 const { block, elem } = bem({
     name: 'Drawer',
     classnames: styles,
-    propsToMods: ['isOpen'],
+    propsToMods: ['isShownAndExpanded', 'isShownAndClosed', 'isExpanded'],
```

Comment:
seems to me that isShownAndExpanded is always equal isExpanded. Can they be merged? Also maybe the naming can be simplified then

Reply:
I changed names for status `IS_EXPANDED`, `IS_CLOSED`, `IS_HIDDEN` and now it looks more clear, thanks. For CSS I would like to keep old names because it reflects real visual status for the Drawer. For example, isExpanded can be not just in shown mode.